core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -961,11 +956,7 @@ protected final void closeNoLock(String reason) {
                 }
                 // no need to commit in this case!, we snapshot before we close the shard, so translog and all sync'ed
                 logger.trace("rollback indexWriter");
-                try {
-                    indexWriter.rollback();
-                } catch (AlreadyClosedException e) {
-                    // ignore
-                }
+                indexWriter.rollback();
                 logger.trace("rollback indexWriter done");
             } catch (Exception e) {
```

Comment:
I didn't think it would?  It's not supposed to?  If it does I think it's a Lucene bug?  We have this comment in `IndexWriter.rollback`:

```
    // don't call ensureOpen here: this acts like "close()" in closeable.
```